docs/improving-gci-experience.md

```diff
@@ -0,0 +1,377 @@
+etc.
+
+* Task functionality - the submission implements the required functionality
+appropriately.
+
+* Code quality and style (as in [Zulip code style
+  doc](http://zulip.readthedocs.io/en/latest/code-style.html)).
+
+* Passing the test suite.
+
+* Passing manual tests.
+
+* Files are properly rendering (especially Markdown files).
+
+* User docs adhere to the
+[style guide](http://zulip.readthedocs.io/en/latest/README.html#style-guide).
+
+We also apply rules described in the [Zulip
+review doc](http://zulip.readthedocs.io/en/latest/code-reviewing.html).
+
+## Why is it taking so long for the mentors to check my task?
+
+With reviewing the GCI tasks we prioritize fairness and transparency. We are
+open to helping students whenever they require assistance, but we also keep
+in mind that both GCI and being a great open-source contributor in general
+is about individual contributions and work. It wouldn't be fair to the
+students who are submitting tasks to get blocked because others might be
+more vocal on chat.
+
+We're trying to keep response times to minimum and one of the more objective
+resources we have is the queue in the GCI app (example screenshot below).
+If you've been working on your task and asking for help on chat, you need to
+submit the task and it will be queued in the app. It's not a perfect solution,
+but keeping with the queue assures we are more objective than by simply
+attending only to students active on Zulip chat.
+
+![Example dashboard view](gci-app-screenshot.png)
+
+Other important points about the response times:
+
+* Mentors are working shifts across multiple timezones, to assure the best
+coverage - if a mentor is online, it doesn't mean they're working on the
+GCI tasks, since we also have our own tasks for Zulip when not on shift.
+
+* The [GCI roles and responsibilities](https://developers.google.com/open-source/gci/help/responsibilities)
+ have some general rules about response times:
+  > Respond to questions within 24 hours (occasionally under 36 hours is ok)
+
+* If it's been more than four hours since you've gotten attention from a
+mentor, it's reasonable to post something on your task topic that says
+you've been waiting for a while. Include useful things like your PR link,
+your task link, and a short status of where things left off with the
+previous mentor.
+
+* Remember that multiple mentors are going to check your work - be sure to
+answer all of their questions and address any issues they point out. Any
+mentor can approve a task in the GCI app if it fulfills the requirements.
+They sometimes need to consult with contributors more experienced in a
+given area, to make sure all the requirements were fulfilled. 
+
+* If you're not getting a quick response from a mentor, take the time to
+double-check your work or find other constructive things to do. Make sure
+you don't have idle time - note down your learning progress, seek out new
+challenges, work on a Python tutorial etc.
```

Comment:
I might add that in any software development community, double-checking your work and making sure it is high quality is the single most valuable thing you can do to minimize the time you're waiting for feedback, since it decreases the number of round trip times you need to wait for feedback.